### PR TITLE
Store Orders: Update status labels for "cash on delivery" orders

### LINKED ISSUE #18955: Store: "Paid in Full" string is confusing when a cash on delivery order is created
When going through the checkout process, if "cash on delivery" option is chosen by the customer, it will show as "Paid in full" automatically from the store owner point of view, as if was paid online. This can be confusing when keeping track of which orders had been paid and which aren't.

This can be seen here: p90Yrv-es-p2 on the minute 32

**Replicate:**
- Go through the process of purchasing a product, you can test is here: https://mug4you.blog
- Go to https://domain.com/checkout or https://mug4you.blog/checkout , fill the data and click on "Place Order"
- Go to http://wordpress.com/store/orders
- See the order as well as "Paid in full"

screen shot 2017-10-18 at 14 31 22

**What I expected:**

- To see some "Pending payment" string or a checkbox to check if this payment has been done

**What happened instead:**

- Is confusing because is a cash on delivery order, just placed it and says it has been paid in full
- Going to the order details shows also how the order (in this case the specific URL is http://wordpress.com/store/orders/mug4you.blog/61) has been already paid:

screen shot 2017-10-18 at 14 34 38

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #18955: Orders placed with "cash on delivery" as payment are sent into the processing status, which we currently label as "Paid in full". This isn't true for COD orders, but they do need to be shipped out, so they're not awaiting payment either. This PR updates the `OrderStatus` component to take the whole `order` object, and check the payment status for the special "COD" case. I've also added tests for this component (thus increasing PR size 😣 ).

New label in context, with other statuses:

screen shot 2017-10-18 at 4 35 28 pm

On the single order screen:

screen shot 2017-10-18 at 4 35 56 pm

And the payment card has also been updated to indicate that payment will happen later, and no refund button (since no payment to refund yet). Could use better copy, maybe?

screen shot 2017-10-18 at 4 54 48 pm

To test

- Run tests: `npm run test-client client/extensions/woocommerce/components/order-status`
- Create a cash on delivery order (you might need to enable the payment option in settings)
- View your orders, see the new order with the new label
- Click into the order, see the payment status label in the top right
- The payment card notes payment on delivery, with no refund option

### CODE REVIEW

File: client/extensions/woocommerce/app/order/order-payment/index.js [hunk 1]
```diff
@@ -63,6 +63,13 @@ class OrderPaymentCard extends Component {
 					refund: formatCurrency( refund, order.currency ),
 				},
 			} );
+		} else if ( 'cod' === order.payment_method && 'processing' === order.status ) {
+			paymentStatus = translate( 'Payment of %(total)s on delivery', {
+				args: {
+					total: formatCurrency( order.total, order.currency ),
+					method: order.payment_method_title,
```
Comment (timmyc) [inline client/extensions/woocommerce/app/order/order-payment/index.js]: Is `method` being used here?

File: client/extensions/woocommerce/app/order/order-payment/index.js [hunk 2]
```diff
@@ -76,7 +83,8 @@ class OrderPaymentCard extends Component {
 
 	getPaymentAction = () => {
 		const { order, translate } = this.props;
-		if ( 'refunded' === order.status ) {
+		const codProcessing = 'cod' === order.payment_method && 'processing' === order.status;
```
Comment (timmyc) [inline client/extensions/woocommerce/app/order/order-payment/index.js]: Duplicate logic from line 86, maybe we should extract it out - or just leave it - 6 one way, half dozen the other 😆

File: client/extensions/woocommerce/components/order-status/test/index.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/**
```
Comment (timmyc) [inline client/extensions/woocommerce/components/order-status/test/index.js]: Very nice!

File: client/extensions/woocommerce/app/order/order-payment/index.js (hunk 2 above)
Comment (ryelle) [reply client/extensions/woocommerce/app/order/order-payment/index.js]: Had the same thought, but then I'd have to write tests for another status function 😆

**Pull request merged**